bobrun/with_init.go

```diff
@@ -227,8 +229,15 @@ func (rw *WithInit) shexec(ctx context.Context, cmds []string) (err error) {
 		p, err := syntax.NewParser().Parse(strings.NewReader(run), "")
 		errz.Fatal(err)
 
-		// FIXME: make run cmds ready for nix integration.
 		env := os.Environ()
+		if len(rw.run.storePaths) > 0 && rw.run.UseNix() {
```

Comment:
Can we move this to a common function? To make sure we use the same logic on `build` & `run` & execctl